libraries/AP_HAL_ChibiOS/EventSource.cpp

```diff
@@ -5,16 +5,17 @@ using namespace ChibiOS;
 
 #if CH_CFG_USE_EVENTS == TRUE
 
-bool EventSource::wait(uint64_t duration, AP_HAL::EventHandle *evt_handle)
+bool EventSource::wait(uint16_t duration_us, AP_HAL::EventHandle *evt_handle)
 {
     chibios_rt::EventListener evt_listener;
     eventmask_t evt_mask = evt_handle->get_evt_mask();
     msg_t ret = msg_t();
     ch_evt_src_.registerMask(&evt_listener, evt_mask);
-    if (duration == 0) {
+    if (duration_us == 0) {
         ret = chEvtWaitAnyTimeout(evt_mask, TIME_IMMEDIATE);
     } else {
-        ret = chEvtWaitAnyTimeout(evt_mask, MAX(CH_CFG_ST_TIMEDELTA, chTimeUS2I(duration)));
+        const sysinterval_t wait = MIN(TIME_MAX_INTERVAL, MAX(CH_CFG_ST_TIMEDELTA, chTimeUS2I(duration_us)));
```

Comment:
I think wait is wrong name to use, its the name of the function as well.

Reply:
fixed